.github/workflows/pages.yml

```diff
@@ -0,0 +1,61 @@
+# GitHub Pages Configuration
+#
+# This workflow builds the repository's website and deploys it using GitHub Pages.
+name: Deploy website with GitHub Pages
+
+on:
+  # Run on all pushes to main.
+  push:
+    branches: ['main']
```

Comment:
What will happen to this config item if the repo's GH pages settings are different? E.g., a different branch or a different folder?

Reply:
@awdeorio I just re-read your question. The repo settings only allow you to choose between deploying a branch or deploying via GitHub Actions. If you choose GitHub Actions, then you cannot specify a branch/folder because that's defined in the GitHub Actions YML config :)